client/src/Models/Ringdown.js

```diff
@@ -200,23 +270,46 @@ class Ringdown {
     const sorted = Object.values(this.validationData).sort(this.ascendingByOrder);
     const previousFields = sorted.slice(0, partition);
     previousFields
-      .filter((fieldData) => fieldData.validationState === ValidationState.NO_INPUT)
+      .filter((fieldData) => fieldData.validationState === ValidationState.EMPTY_REQUIRED_INPUT)
       .forEach((fieldData) => {
-        this.validationData[fieldData.name].validationState = ValidationState.ERROR;
+        this.validationData[fieldData.name].validationState = ValidationState.REQUIRED_ERROR;
       });
   }
 
   setValidationStateForInput(fieldName, currentState, inputValue) {
     const isInputValueEmpty = isValueEmpty(inputValue);
-
-    if (currentState === ValidationState.ERROR && !isInputValueEmpty) {
-      this.validationData[fieldName].validationState = ValidationState.FIXED;
-    } else if (currentState === ValidationState.NO_INPUT) {
-      this.validationData[fieldName].validationState = ValidationState.INPUT;
-    } else if (currentState === ValidationState.INPUT && isInputValueEmpty) {
-      this.validationData[fieldName].validationState = ValidationState.ERROR;
-    } else if (currentState === ValidationState.FIXED && isInputValueEmpty) {
-      this.validationData[fieldName].validationState = ValidationState.ERROR;
+    const { range, required, allowNull } = Patient[fieldName] || Ambulance[fieldName] || EmergencyMedicalServiceCall[fieldName] || {};
```

Comment:
I don't think checking `allowNull` is required, er. needed here.  I think that value is only used by sequelize, and is separate from whether the field must be filled in by the user.

Reply:
The reason I also included allowNull is because certain required inputs were not throwing the expected error when empty and required (see dispatchCallNumber and ambulanceIdentifier) because they don't have a required field. I could add a required field in the metadata to the ones missing if you think that would make things more clear.

Reply:
Ah, you're right.  Fields in PatientFields that are created via a `FormField`, like age, pull the `required` value from metadata.  But a few don't, like the unit ID, since it was a little harder to create those form elements automatically.  And those specify `required` directly in the component, even if it's not in the metadata.

Please do add the missing required flags to the metadata, so we can remove the `allowNull` checks.  We can look at improving that name, as there are fields that can't be null in the database, but aren't visible to the user, so they're not "required" in that sense.

Reply:
Sounds good. Should i remove the allowNull fields from the metadata or keep them and just add a required field?

Reply:
Leave the allowNull fields, as they matter to sequelize.  We just need to add the `required` flag to a few of the fields that are required to be filled out in the form.